h2o-admissibleml/src/main/java/hex/Infogram/InfogramUtils.java

```diff
@@ -105,12 +110,25 @@ public static Frame extractTrainingFrame(InfogramParameters parms, String[] sens
         extractedFrame.add(colName, trainFrame.vec(colName));
 
     List<String> colNames = Arrays.asList(trainFrame.names());
-    if (parms._weights_column != null && colNames.contains(parms._weights_column))
-      extractedFrame.add(parms._weights_column, trainFrame.vec(parms._weights_column));
+    boolean cvWeightsPresent = parms._weights_column != null && colNames.contains(parms._weights_column)
+            && parms._weights_column.equals("__internal_cv_weights__");
+    if (parms._weights_column != null && colNames.contains(parms._weights_column)) {
```

Comment:
@maurever 

In infogram, other algos are called to build models.  Infogram info is derived from these models.

When CV is enabled, infogram will run in multiple folds.  However, for each fold, infogram will call other algos to build models for it.  However, it will disable CV for those algos.  In order to do this, infogram will need to do multiple things to create this illusion for the other models.

Here is the new code:

      String[] nonPredictors = parms.getNonPredictors();
    List<String> colNames = Arrays.asList(trainFrame.names());
      boolean cvWeightsPresent = parms._weights_column != null && colNames.contains(parms._weights_column)
            && parms._weights_column.equals("__internal_cv_weights__");
      for (String nonPredName : nonPredictors) {
        if ("__internal_cv_weights__".equals(nonPredName) && colNames.contains(parms._weights_column)) {
          String cvWeightName = "infogram_internal_cv_weights_"; // switch weights column to turn off cv in algo used to build infogram
          extractedFrame.add(cvWeightName, trainFrame.vec(nonPredName));
          parms._weights_column = cvWeightName;
        } else if (nonPredName.equals(parms._fold_column) && colNames.contains(parms._fold_column) && !cvWeightsPresent) {
          extractedFrame.add(nonPredName, trainFrame.vec(nonPredName));
        } else {
          extractedFrame.add(nonPredName, trainFrame.vec(nonPredName));
        }
      }

Inside each fold, Infogram needs to know that CV is enabled.  This is done by checking is parms._weight_column.equals("__internal_cv_weights__").

If cvWeight is detected, infogram need to add the weight column to the training frame that is going to be passed to other algos.  However, it needs to tell the other algo that there is no CV.  To do that, it will add the weights to the training frame but will change the weight name from "__internal_cv_weights__" to "infogram"__internal_cv_weights__".

You have question on this statement:
if (nonPredName.equals(parms._fold_column) && colNames.contains(parms._fold_column) && !cvWeightsPresent)

this condition checks that fold_column exists and there is no CV weight column: "__internal_cv_weights__" exist.  If the user specify a weight column as say parms._weight_column = "my_weight", the fold_column will be added to the training frame which is passed to the other algos.  So, the code did not say that you cannot specify a fold_column and a weight column.

Does that answer your question?

Reply:
Yes, thank you for the explanation!